libraries/AP_DDS/AP_DDS_External_Odom.cpp

```diff
@@ -0,0 +1,60 @@
+
+
+#if AP_DDS_ENABLED && defined(HAL_VISUALODOM_ENABLED)
+
+#include "AP_DDS_External_Odom.h"
+#include <AP_VisualOdom/AP_VisualOdom.h>
+
+void AP_DDS_External_Odom::handle_external_odom(const tf2_msgs_msg_TFMessage& msg)
+{
+    AP_VisualOdom *visual_odom = AP::visualodom();
+    if (visual_odom != nullptr) {
+        for (size_t i = 0; i < msg.transforms_size; i++) {
+            const auto ros_transform_stamped = msg.transforms[i];
+            if (is_odometry_frame(ros_transform_stamped)) {
+                const uint64_t remote_time_us {0}; // TODO
```

Comment:
remote time can be any time base. Populate this with the message timestamp. In mavlink, it's a time in microseconds. It doesn't matter, the jitter corrector will handle it. 

time_t
* seconds uint64_t (linux epoch)
* milliseconds uint32 up to 1 second. 

Needs converted to single value.